includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -104,6 +105,33 @@ public function sanitize() {
 
 			$this->process_gallery_embed( $is_amp_carousel, $is_amp_lightbox, $gallery_element, $img_elements );
 		}
+
+		if ( ! isset( $this->args['native_img_used'] ) || ! $this->args['native_img_used'] ) {
+			return;
+		}
+
+		$img_elements = $this->dom->xpath->query(
+			'.//img[ @layout = "fill" or @object-fit = "cover" ]',
+			$this->dom->body
+		);
+
+		/*
+		* Remove `layout` and `object-fit` attributes from native img as they are not supported by AMP yet.
+		* Add style attribute which is used to set `object-fit` and `layout` attributes.
+		* @see <https://github.com/ampproject/amp-wp/issues/7152#issuecomment-1157933188>
+		* @todo Remove this once `layout` and `object-fit` attributes are supported by AMP spec for native img tag.
+		*/
+		$style_layout_fill = 'position:absolute; left:0; right:0; top:0; bottom: 0; width:100%; height:100%;';
+		$style_object_fit  = 'object-fit:cover;';
+		foreach ( $img_elements as $img_element ) {
+			$remove_layout_attr     = $img_element->removeAttribute( Attribute::LAYOUT );
+			$remove_object_fit_attr = $img_element->removeAttribute( Attribute::OBJECT_FIT );
+			$style_attr_content     = sprintf( '%s %s', $remove_layout_attr ? $style_layout_fill : '', $remove_object_fit_attr ? $style_object_fit : '' );
+
+			if ( ' ' !== $style_attr_content ) {
+				$img_element->setAttribute( Attribute::STYLE, $style_attr_content );
+			}
+		}
```

Comment:
I think this logic may make more sense in the image sanitizer, specifically in the `adjust_and_replace_node` method:

https://github.com/ampproject/amp-wp/blob/bbf48d73086f64dc7b569199ca9e55e5937225fd/includes/sanitizers/class-amp-img-sanitizer.php#L402-L422

Reply:
Where you're currently adding the `lightbox` handling.

Reply:
> Where you're currently adding the lightbox handling.

Handling it in the image sanitizer because need to check if an image has `lightbox` enabled to not.

https://github.com/ampproject/amp-wp/blob/60e2f46b9bfc573bb11e394877bcc63fa655a486/includes/sanitizers/class-amp-img-sanitizer.php#L405-L418

Reply:
I mean, I think the logic here would all make more sense in the image sanitizer.

Reply:
Oh, but the problem is that `AMP_Gallery_Block_Sanitizer` runs after `AMP_Img_Sanitizer`.

Reply:
I would look into changing the order of `$expected_final_sanitizer_order` so that `AMP_Gallery_Block_Sanitizer` runs before `AMP_Img_Sanitizer`.

Reply:
IMO we need this logic in both image and gallery sanitizer. 

Reason for gallery sanitizer:
As far as I have explored this sanitizer I have observed that the carousel is constructed in gallery sanitizer which runs after image sanitizer. When the carousel is constructed it adds the `layout` and `object-fit` attrs.
https://github.com/ampproject/amp-wp/blob/bbf48d73086f64dc7b569199ca9e55e5937225fd/src/Component/Carousel.php#L92-L102

Reason for image sanitizer
Maybe `layout` and `object-fit` present in the image tag to needs to be sanitized.

Reply:
I think I should the same logic for replacing `layout` and `object-fit` attrs in the image sanitizer as well.

Reply:
If the gallery sanitizer could be run first, then it seems the logic could be just kept in the img sanitizer. This will ensure that anything that adds `layout` or `lightbox` to `img` elements (whether by our sanitizers or by someone else) will then get processed by the img sanitizer.

Reply:
Yes. That logic can be added to the img sanitizer's `adjust_and_replace_node` method, which processes over every `img` in the document.

Reply:
> If the gallery sanitizer could be run first, then it seems the logic could be just kept in the img sanitizer. 

It would be great if we change the order of sanitizer runs but I hope it doesn't break anything. BTW I think we need to remove this logic as well when things will be amended to AMP spec.

Reply:
I believe the only logic no longer applicable once the AMP spec is updated is marking the `layout` attribute as being PX-verified. Everything else will remain, including the `object-fit` and `layout` conversions which won't be applicable to `img`.

Reply:
Oh! that's the thing. I extracted the idea of AMP spec from the above conversations and do not have many ideas about it. Any resource from where I can learn about this AMP spec?

Reply:
Here are a couple resources:

* https://amp.dev/documentation/guides-and-tutorials/learn/validation-workflow/validate_amp/?format=websites
* https://amp.dev/documentation/guides-and-tutorials/learn/validation-workflow/validation_errors/?format=websites

The validator spec is broken up into various protoascii files:

* https://github.com/ampproject/amphtml/blob/main/validator/validator-main.protoascii
* https://github.com/ampproject/amphtml/blob/main/validator/validator-css.protoascii
* https://github.com/ampproject/amphtml/blob/main/validator/validator-svg.protoascii

And each extension has its own protoascii spec, for example: https://github.com/ampproject/amphtml/blob/main/extensions/amp-base-carousel/validator-amp-base-carousel.protoascii

The AMP plugin consumes this spec to make the massive PHP file that we use for validation.

Reply:
Are you still planning to try to move this code to the img sanitizer by changing the order that the sanitizers run in? That is, if the gallery block sanitizer runs before the image sanitizer then the logic should be able to be consolidated in the img sanitizer.

Reply:
> Are you still planning to try to move this code

I have tried it locally but changing the order of sanitizers causes markup error as the gallery block expects sanitized images. 

One more idea that came up to my mind is to create a mini native IMG attributes sanitizer which can be called after gallery block. Doing so can help to sanitize more attributes if required in near future.

Reply:
I do like your idea of having a new mini native img sanitizer that runs after the gallery sanitizer to specifically handle the `lightbox`, `layout`, and `object-fit` attributes. It would be better yet to have it run in the img sanitizer, but consolidating them into one special sanitizer is better than having them spread out across two sanitizers as they are now.

Reply:
So, should I create a mini native img attributes sanitizer?

Reply:
Yes, please create an `AMP_Native_Img_Attributes_Sanitizer`.